src/core/segment_reader.rs

```diff
@@ -287,8 +287,19 @@ impl SegmentReader {
     }
 
     /// Returns an iterator that will iterate over the alive document ids
-    pub fn doc_ids_alive(&self) -> impl Iterator<Item = DocId> + '_ {
-        (0u32..self.max_doc).filter(move |doc| !self.is_deleted(*doc))
+    pub fn doc_ids_alive(&self) -> Box<dyn Iterator<Item = DocId> + '_> {
+        if let Some(delete_bitset) = &self.delete_bitset_opt {
+            Box::new(
+                delete_bitset
+                    .iter()
+                    .take(self.max_doc() as usize)
```

Comment:
It's now part of the iterator and not the api

Each block contains 64 elements, but we only want to iterate until max_doc, since the last elements will be emitted as alive